opengever/maintenance/scripts/list_dossiers_shared_with_group.py

```diff
@@ -0,0 +1,92 @@
+"""
+Searches for all resolved dossiers, on which local_roles were set for a given
+group via sharing.
+
+    bin/instance run ./scripts/list_dossiers_shared_with_group.py
+
+"""
+from opengever.base.role_assignments import ASSIGNMENT_VIA_SHARING
+from opengever.base.role_assignments import RoleAssignmentManager
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from opengever.maintenance.utils import LogFilePathFinder
+from opengever.maintenance.utils import TextTable
+from plone import api
+import sys
+
+
+class DossierSharedWithGroupLister(object):
+
+    def __init__(self, group):
+        self.table = TextTable(col_max_width=60)
+        self.table.add_row(["dossier path", "dossier title", "roles"])
+        self.group = group
+
+    def list_dossiers_shared_with_group(self):
+        for dossier, assignment in self.get_dossiers_shared_with_group():
+            self.table.add_row([dossier.absolute_url_path(), dossier.Title(), assignment.roles])
+
+    def print_table(self):
+        print("Table of dossiers shared with {}".format(self.group.id))
+        print(self.table.generate_output())
+        print("\nSummary:")
+        print("There are {} dossiers shared with {}".format(self.table.nrows,
+                                                            self.group.id))
+
+    def get_dossiers_shared_with_group(self):
+        """ Searches for all dossier shared with a given group.
+        We make the assumption that the group has at least view permissions
+        on such dossiers.
+        """
+        dossier_brains = api.content.find(allowedRolesAndUsers=u'user:{}'.format(self.group.id))
+        for dossier_brain in dossier_brains:
+            dossier = dossier_brain.getObject()
+            assignments = RoleAssignmentManager(dossier).get_assignments_by_principal_id(self.group.id)
+            sharing_assignment = self._find_sharing_assignment(assignments)
+            if sharing_assignment:
+                yield (dossier, sharing_assignment)
+        return
+
+    @staticmethod
+    def _find_sharing_assignment(assignments):
+        try:
+            index = map(lambda assignment: assignment.cause, assignments).index(ASSIGNMENT_VIA_SHARING)
```

Comment:
i find this much harder to read than something like

```python
sharing_assignments = [each for each in assignments if each.cause == ASSIGNMENT_VIA_SHARING]
```

also, could there be multiple assignments? probably not, right 🤔?

Reply:
yes there could be multiple assignments. Maybe one from sharing and one from a forwarding for example

Reply:
But only one from assignment. That's why I look for the index.

Reply:
yes per category and user i meant. but don't see how either.